pkg/kademlia/dialer.go

```diff
@@ -109,6 +109,26 @@ func (dialer *Dialer) FetchPeerIdentity(ctx context.Context, target pb.Node) (pI
 	return identity.PeerIdentityFromPeer(p)
 }
 
+// FetchInfo connects to a node address and returns its node info.
+func (dialer *Dialer) FetchInfo(ctx context.Context, address *pb.NodeAddress) (*identity.PeerIdentity, *pb.InfoResponse, error) {
+	if !dialer.limit.Lock() {
+		return nil, nil, context.Canceled
+	}
+	defer dialer.limit.Unlock()
+
+	conn, err := dialer.dialAddress(ctx, address)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	p := &peer.Peer{}
+	pCall := grpc.Peer(p)
+	resp, err := conn.client.RequestInfo(ctx, &pb.InfoRequest{}, pCall)
+	id, idErr := identity.PeerIdentityFromPeer(p)
```

Comment:
Should we do this step if `RequestInfo` returns error?

Reply:
You are right. Fixed it in the latest commit.